components/codec/src/byte.rs

```diff
@@ -112,6 +159,17 @@ impl MemComparableByteCodec {
         }
     }
 
+    /// Performs in place bitwise NOT for specified memory region, without doing bounds checking.
+    #[inline]
+    fn flip_bytes_in_place_unchecked(src: &mut [u8], len: usize) {
```

Comment:
I dont think this function is very necessary :) Could you post a benchmark to compare it with the checked version? I believe rust compiler will check bound only once for the checked version.

Reply:
This function is for my previous unchecked implementation and now is not necessary :)